bin/alluxio-start.sh

```diff
@@ -515,11 +515,12 @@ main() {
       start_proxies
       ;;
     local)
-      ALLUXIO_MASTER_JAVA_OPTS+=" -Dalluxio.master.hostname=localhost"
-      ALLUXIO_WORKER_JAVA_OPTS+=" -Dalluxio.master.hostname=localhost"
-      ALLUXIO_PROXY_JAVA_OPTS+=" -Dalluxio.master.hostname=localhost"
-      ALLUXIO_JOB_MASTER_JAVA_OPTS+=" -Dalluxio.master.hostname=localhost"
-      ALLUXIO_JOB_WORKER_JAVA_OPTS+=" -Dalluxio.master.hostname=localhost"
+      local hostname=$(${BIN}/alluxio getConf alluxio.master.hostname)
+      if [[ "${hostname}" != "localhost" \
+        && "${hostname}" != $(hostname -s) \
```

Comment:
I would add logic to support 'hostname -i' for providing a direct IP. This should only happen on Linux based machines and not on a Mac.

Reply:
done. PTAL

Reply:
LGTM